source/src/in/joind/DateHelper.java

```diff
@@ -16,13 +17,14 @@
      */
     public static String parseAndFormat(String input, String format) {
         // Assume that the input format is ISO8601
-        SimpleDateFormat dfOutput = new SimpleDateFormat(format), dfInput = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ");
+        SimpleDateFormat dfOutput = new SimpleDateFormat(format, Locale.US),
```

Comment:
Yes, only against the incoming date format.
“The default locale is not appropriate for machine-readable output. The best choice there is usually Locale.US – this locale is guaranteed to be available on all devices” -- http://developer.android.com/reference/java/util/Locale.html